ui/packages/components/src/CancelRunButton/CancelRunModal.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+import { Alert } from '@inngest/components/Alert';
+import { AlertModal } from '@inngest/components/Modal';
+
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+  onSubmit: () => Promise<void>;
+};
+
+export function CancelRunModal({ isOpen, onClose, onSubmit }: Props) {
+  const [error, setError] = useState<string>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  async function handleSubmit() {
+    setIsLoading(true);
+    try {
+      await onSubmit();
+      setError(undefined);
+    } catch (error) {
+      if (!(error instanceof Error)) {
+        setError('Unknown error');
+        return;
+      }
+
+      setError(error.message);
+      throw error;
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  return (
+    <AlertModal
+      className="w-1/3"
+      isLoading={isLoading}
+      isOpen={isOpen}
+      onClose={onClose}
+      onSubmit={handleSubmit}
+      title="Are you sure you want to cancel this function run?"
+    >
+      <p className="pt-4 dark:text-white">
+        The function run will end early and its status will be "cancelled". This action cannot be
```

Comment:
Not for now, but this makes it very evident that we should address the UK vs US en issue with cancelled. It just looks weird